atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/TableDefinition.java

```diff
@@ -62,34 +62,86 @@ public void constraints() {
         state = State.DEFINING_CONSTRAINTS;
     }
 
+    /**
+     * Indicates that the name of the table being defined is safe for logging.
+     */
+    public void tableNameIsSafeLoggable() {
```

Comment:
Yep, agree that the superset option makes sense.

There's an interesting idea here for `schemaSafeForLoggingByDefault()` or something like that, that makes the default for **everything** safe unless otherwise indicated.

Might include this in the next PR though.